### PR TITLE
treewide: remove outdated multiple versions of Qt stuff

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

Hopefully we won't have to do this anymore. Also clean up all-packages.nix a bit while I'm at it.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by K900** [comment]
CC @alyssais @NixOS/qt-kde

File: doc/languages-frameworks/qt.section.md [hunk 1]
```diff
@@ -2,14 +2,9 @@
...
-There are two problems that the Nixpkgs Qt infrastructure addresses,
-which are not shared by other C++ software:
 
-1.  There are usually multiple supported versions of Qt in Nixpkgs.
-    All of a package's dependencies must be built with the same version of Qt.
-    This is similar to the version constraints imposed on interpreted languages like Python.
-2.  Qt makes extensive use of runtime dependency detection.
-    Runtime dependencies are made into build dependencies through wrappers.
+However, Qt makes extensive use of runtime dependency detection.
+In nixpkgs, those runtime dependencies are made into build dependencies through wrappers.
```
Comment (samueldr) [inline doc/languages-frameworks/qt.section.md:7]: If I didn't know already, I would think *made into build dependencies* here means that the build will error out when runtime dependencies are missing during build-time.

Is there a way we can rephrase and signal correctly that this is adds *tracking* for runtime dependencies at build-time?

Comment (K900) [reply doc/languages-frameworks/qt.section.md]: How's this?

**Comment by FRidh** [comment]
> Hopefully we won't have to do this anymore. 

I haven't been keep up lately but how is this outdated information?

**Comment by K900** [comment]
We don't have multiple Qt5 versions anymore.

**Pull request merged**